### PR TITLE
Admin Menu: Mimic existing Calypso menu on Jetpack sites

### PR DESCRIPTION
**Pull request by mmtr**
Fixes (partially) https://github.com/Automattic/wp-calypso/issues/50438

#### Changes proposed in this Pull Request:
Implements some enhancements to improve the UX of the new unified Calypso menu on Jetpack sites:
- Update `Jetpack_Admin_Menu` so we don’t return more items that the ones currently present in the existing navigation.
- Make sure all links point to Calypso (regardless of the link destination toggle). There are a few exception for this, since there are some screens that are not supported by Calypso (Feedback, Customize, Import, Export, WP Admin).

This will be followed up with a Calypso PR that will display an external icon on the WP Admin exceptions while ensuring they will be opened on a new tab: https://github.com/Automattic/wp-calypso/pull/50444

Existing nav | New unified nav (*)
--- | ---
Screen Shot 2021-02-24 at 17 24 10 | Screen Shot 2021-02-24 at 17 26 33

(*) I manually expanded all the section to make visible all the submenu. In practice, only one section can be expanded at the same time.

#### Jetpack product discussion
p1HpG7-bcl-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
- Spin up a Jetpack site running this branch.
- Go to wordpress.com and switch to that site.
- Make sure the menu items match with what's in the existing navigation (you can switch to that by adding `?disable-nav-unification` to the URL).
- Install some third party plugins and make sure their menus don't appear in the new unified menu.
- (Optional): Test how this looks in combination with https://github.com/Automattic/wp-calypso/pull/50444.

#### Proposed changelog entry for your changes:
N/A.

### CODE REVIEW

**Review by cpapazoglou — APPROVED** [review]
This one works as expected for me and in combination with https://github.com/Automattic/wp-calypso/pull/50444 shows the external icons too 🎉!

**Review by jeherve — DISMISSED** [review]
This tests well for me. Merging.

**Review by jeherve — DISMISSED** [review]
Re-approving after an update to solve merge conflicts.

**Comment by mmtr** [comment]
Looking at the failing test 👀

File: projects/plugins/jetpack/tests/php/core-api/wpcom-endpoints/test-class-wpcom-rest-api-v2-endpoint-admin-menu.php
Comment (mmtr) [inline projects/plugins/jetpack/tests/php/core-api/wpcom-endpoints/test-class-wpcom-rest-api-v2-endpoint-admin-menu.php:95]: The Settings menu no longer returns submenu items for Jetpack sites, so I had to update this test to user another menu with submenus.

**Review by jeherve — APPROVED** [review]

**Comment by jeherve** [comment]
Cherry-picked to `jetpack/branch-9.5` in 6ab352126e8a044be0b770dba1a91a2a3a81a479

**Comment by mmtr** [comment]
Deployed to WP.com in r221680-wpcom

**Pull request merged**
